tools/verify_safety.py

```
@@ -217,14 +221,77 @@ def verify() -> None:
+        fail("version.txt must contain MAJOR.MINOR.PATCH")
+
+    assembly_info = ASSEMBLY_INFO.read_text(encoding="utf-8")
+    assembly_version = version + ".0"
+    if '[assembly: AssemblyVersion("' + assembly_version + '")]' not in assembly_info:
+        fail("AssemblyVersion is not synchronized with version.txt")
+    if '[assembly: AssemblyFileVersion("' + assembly_version + '")]' not in assembly_info:
+        fail("AssemblyFileVersion is not synchronized with version.txt")
+
+    # SubModule.xml is trusted repository input. Standard-library parsing keeps
+    # this verifier hermetic and does not add a network-installed dependency.
     module = ET.fromstring(MODULE_XML.read_text(encoding="utf-8"))
```

Ruff (0.15.20)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)